azure-cs-credential-rotation-one-set/AppStack.cs

```diff
@@ -0,0 +1,315 @@
+                    },
+                }
+                },
+                EnabledForDeployment = false,
+                EnabledForDiskEncryption = false,
+                EnabledForTemplateDeployment = false,
+                Sku = new SkuArgs
+                {
+                    Family = "A",
+                    Name = KeyVault.SkuName.Standard,
+                },
+                TenantId = tenantId,
+                EnableSoftDelete = false,
+            },
+            ResourceGroupName = resourceGroupName,
+            VaultName = Output.Format($"{resourceNamePrefix}-kv"),
+        });
+
+        var topicName = "SecretExpiry";
+        var topic = new SystemTopic("eventGridTopic",
+            new SystemTopicArgs
+            {
+                SystemTopicName = topicName,
+                Source = keyVault.Id,
+                TopicType = "microsoft.keyvault.vaults",
+                Location = resourceGroup.Apply(r => r.Location),
+                ResourceGroupName = resourceGroup.Apply(r => r.Name),
+            });
+
+        var eventSubscription = new SystemTopicEventSubscription("eventSubscription",
+            new SystemTopicEventSubscriptionArgs
+            {
+                EventSubscriptionName = Output.Format($"{keyVault.Name}-{secretName}-{functionAppName}"),
+                SystemTopicName = topic.Name,
+                ResourceGroupName = resourceGroup.Apply(r => r.Name),
+                Filter = new EventSubscriptionFilterArgs
+                {
+                    SubjectBeginsWith = secretName,
+                    SubjectEndsWith = secretName,
+                    IncludedEventTypes = { "Microsoft.KeyVault.SecretNearExpiry" },
+                },
+                Destination = new AzureFunctionEventSubscriptionDestinationArgs
+                {
+                    ResourceId = Output.Format($"{functionApp.Id}/functions/AKVSQLRotation"),
+                    EndpointType = "AzureFunction",
+                    MaxEventsPerBatch = 1,
+                    PreferredBatchSizeInKilobytes = 64,
+                },
+            },
+            new CustomResourceOptions { DependsOn = { functionAppSourceControl } });
+
+        var expiresAt = DateTimeOffset.Now.AddMinutes(1).ToUnixTimeSeconds();
+        var secret = new Secret("secret",
+            new SecretArgs
+            {
+                SecretName = secretName,
+                VaultName = keyVault.Name,
+                ResourceGroupName = resourceGroupName,
+                Tags = { { "CredentialId", sqlAdminLogin }, { "ProviderAddress", sqlServer.Id }, { "ValidityPeriodDays", "1" } },
+                Properties = new SecretPropertiesArgs
+                {
+                    Value = sqlAdminPassword,
+                    Attributes = new SecretAttributesArgs { Expires = (int)expiresAt },
+                },
+            },
+            new CustomResourceOptions { DependsOn = { eventSubscription } }); 
+
+        #region WebApp
```

Comment:
Nit: we prefer not to use regions in our code

Reply:
Removed.